### PR TITLE
fix(client): dynamic rejectOnNotFound types

### LINKED ISSUE #5190: Narrow type for `rejectOnNotFound`
As of now, the return type of `findFirst` is still `User | null`, even when `rejectOnNotFound` is used.
We should narrow the type down to only `User`.

### PR DESCRIPTION
**Pull request by williamluke4**
closes https://github.com/prisma/prisma/issues/5190

### CODE REVIEW

File: src/packages/client/src/generation/TSClient/PrismaClient.ts [hunk 1]
```diff
@@ -161,7 +164,25 @@ get ${methodName}(): Prisma.${m.model}Delegate;`
 export type RejectOnNotFound = boolean | ((error: Error) => Error)
 export type RejectPerModel = { [P in ModelName]?: RejectOnNotFound }
 export type RejectPerOperation =  { [P in "findUnique" | "findFirst"]?: RejectPerModel | RejectOnNotFound } 
-
+type isTrue<T> = T extends true ? 1 : T extends (err: Error)=> Error ? 1 : 0
```
Comment (timsuchanek) [inline src/packages/client/src/generation/TSClient/PrismaClient.ts:167]:
```suggestion
type isTrue<T> = T extends true ? 1 : T extends (err: Error) => Error ? 1 : 0
```

Comment (timsuchanek) [reply src/packages/client/src/generation/TSClient/PrismaClient.ts:167]: And we should call `isTrue` `IsTrue` - uppercase. That's a convention we have in our codebase to distinguish types from the rest.

Comment (williamluke4) [reply src/packages/client/src/generation/TSClient/PrismaClient.ts:167]: Changing it from isTrue to IsReject

**Comment by dickfickling** [comment]
@timsuchanek  any chance we could get a patch release soon with this? really looking forward to cleaning up all my guard code!

**Pull request merged**